uwblueprint/starter-code-v2
# Add files and modifications for backend deployment to Heroku

## Notion ticket link

N/A



## Implementation description
* Make backend deployable to Heroku by removing dependency on gitignored files (Heroku deployment requires pushing to a git remote)
    * `nodemailer.config.ts`: now reads secrets through `process.env` instead of using raw values
    * `firebaseServiceAccount.json`: deleted, Firebase Admin SDK initialization modified so that it no longer requires this file
 * Added required deployment files, e.g. `Procfile` and build scripts
 * Added GitHub Actions for automated Heroku deployment
 * Require using migrations in TypeScript backend instead of using Sequelize.sync()

Based on: https://github.com/uwblueprint/sistering/pull/66 and https://github.com/uwblueprint/planet-read/pull/67



## Steps to test
1. Update root `.env` file
2. Remove old volumes
```
docker-compose down --volumes
```
3. Run the application using the TypeScript and Python backends separately (drop all tables between runs as the two backends have slightly different DB schemas), remember to run migrations:
    a. TypeScript: `docker exec -it scv2_ts_backend -c "node migrate up"`
    b. Python:  `docker exec -it scv2_py_backend -c "flask db upgrade"`
4. For each of the backends, check that refactored code paths have **not** resulted in changed behaviour:
    a. Ensure there are no errors in Firebase initialization
    b. Login, then send a reset password email `http://localhost:5000/auth/resetPassword/<YOUR-EMAIL>`
    c. Fetch an existing file: `http://localhost:5000/entities/files/2165d519-c270-482b-b57b-c3583e9ac2b9`
5. The backend deployment GitHub Actions are currently disabled so cannot directly verify correctness, however they are almost exact copies of the Actions I wrote for Sistering which work: https://github.com/uwblueprint/sistering/pull/66


## What should reviewers focus on?
* Are there any other required deployment changes that I missed?


## Checklist
